dcpy/lifecycle/package/validate.py

```diff
@@ -76,20 +77,23 @@ def _is_valid_wkb(g):
         return False
 
 
-def _is_float_or_double(s):
-    try:
-        float(s)
-        return True
-    except ValueError:
-        return False
-
-
 def _is_int(s):
     if s[0] in ("-", "+"):
         return s[1:].isdigit()
     return s.isdigit()
 
 
+_decimal_regex = re.compile(r"^-?\d*\.\d*$")
+
+
+def _is_decimal(s):
+    return bool(_decimal_regex.match(s)) if s else False
+
+
+def _is_number(s):
```

Comment:
What's the rational for having ints, decimals, and numbers?

Reply:
So IIRC we'd settled on `ints` and `decimans` in our talks about ingest validation. I added number just for the case when we don't know or care whether there's a decimal in the number. IIRC it was due to laziness on my part about validating something in the test case. I'm kind of ambivalent about keeping it - open to nixing it. Also looping in @damonmcc and @sf-dcp for thoughts.

Reply:
Gotcha - makes sense on the validation side, though in terms of the actual data I think `decimal` is fine as a catch-all for numbers. Counter argument might still be that we want our metadata to be as descriptive as possible of the actual structure of the data, but as a user I would prefer to just see "decimal" even if it's a mix of floats and ints

Reply:
That's interesting... one of the reasons for the current implementation is to distinguish between, say, "10" and "10.0", which touches on both formatting and underlying implementations. I'm wondering if it makes sense to distinguish between those, or allow them both to be decimals. It seemed vaguely important when I wrote this, but now I'm not so sure.

Reply:
I'm down to drop `number` so that while validating we're forced to learn (and then accurately declare) the type, but I'd understand if you wanna keep the flexibility for now

and since columns usually can't (and shouldn't?) have mixed types, if a column has both "9.5" and "10" in it it'd expect to see/use them as `["9.5", "10.0"]`

Reply:
agree w/@damonmcc